### PR TITLE
Add paged attention decode with Gluon JIT/AOT for AMD CDNA3

### PR DESCRIPTION
**Pull request by yanguahe**
### Motivation

This PR adds high-performance paged attention decode kernels optimized for AMD CDNA3 architecture (MI300 series GPUs) using Triton's Gluon framework. The implementation addresses the need for:

1. **Efficient Long-Context Inference**: Paged attention mechanism enables handling long sequences by efficiently managing KV cache in memory-efficient blocks
2. **FP8 Quantization Support**: Native support for FP8 data types reduces memory bandwidth requirements and improves performance on AMD MI300 GPUs
3. **AMD GPU Optimization**: Leverages AMD MFMA (Matrix Fused Multiply-Add) instructions and CDNA3-specific memory layouts for optimal hardware utilization
4. **AOT Compilation**: Provides ahead-of-time compilation infrastructure for deployment scenarios requiring pre-compiled kernels

### Technical Details

This PR introduces a comprehensive paged attention decode implementation with the following components:

**Core Kernel Implementation** (`aiter/ops/triton/gluon/pa_decode_gluon.py`):
- Two main kernel variants:
  - `paged_attention_decode_v2_gluon_fp8`: Optimized for standard block sizes (16, 64)
  - `paged_attention_decode_v2_gluon_large_block_fp8`: Optimized for large block size (1024)
- Reduction kernels for combining partial attention results from multiple context partitions
- Support for both Gluon JIT and standard Triton modes

**Key Features**:
- **Quantization Modes**: Both per-token and per-tensor quantization for queries and KV caches
- **Flexible Block Sizes**: Support for 16, 64, and 1024 token blocks in paged KV cache
- **Memory Layouts**: Optimized blocked and distributed linear layouts for efficient MFMA instruction usage
- **Numerical Stability**: Numerically stable softmax computation with online maximum tracking
- **Causal Masking**: Built-in support for autoregressive generation patterns
- **Value Cache Transpose**: Optional transposed value cache layout for better memory access patterns

**AOT Compilation Infrastructure** (`csrc/cpp_itfs/pa_gluon_aot/` and `csrc/cpp_itfs/gluon_aot_tools/`):
- Python-based AOT compilation pipeline for Gluon kernels
- C++ template files (Jinja2) for kernel wrapper generation
- Pre-build scripts for generating multiple kernel configurations
- HIP integration for AMD GPU deployment

**Architecture-Specific Optimizations**:
- AMD MFMA layout with [16x16] instruction shapes
- Swizzled shared memory layouts to reduce bank conflicts
- Buffer load/store operations optimized for CDNA3
- Configurable waves-per-EU for occupancy tuning
- Register allocation strategies tailored to different query group sizes (16, 32, 64)

### Test Plan

The PR includes comprehensive testing infrastructure:

1. **Unit Tests** (`op_tests/triton_tests/test_pa_decode_gluon.py`):
   - Parameterized tests covering various configurations:
     - Compute types: FP8, FP16, BF16
     - Block sizes: 16, 64, 1024
     - Head configurations: (5, 1), (8, 1), (10, 1), (16, 1)
     - Context lengths: 256, 512, 1024, 4096, 8192
     - Query lengths: 1, 2, 3, 4
     - Batch sizes: 4, 128
     - Head dimensions: 64, 128, 256
   - Quantization mode testing: per-token and per-tensor
   - Value cache layout variants: standard and transposed
   - Variable-length sequence support

2. **Kernel-Specific Tests**:
   - `pa_attention_kernel_test.py`: Tests for main attention computation kernel
   - `pa_reduce_kernel_test.py`: Tests for reduction kernel
   - Comparison against PyTorch reference implementation
   - Numerical accuracy validation with configurable tolerance (5e-2 for FP8)

3. **AOT Compilation Tests**:
   - Validation of compiled kernels against JIT versions
   - Performance benchmarking infrastructure
   - MD5 hash verification for deterministic outputs

### Test Result

The implementation has been validated with:

- **Correctness**: All test cases pass with max difference < 5e-2 tolerance for FP8 operations
- **Performance**: Achieves competitive bandwidth utilization compared to assembly-optimized kernels
- **Coverage**: Tested across 1000+ parameter combinations (with configurable sampling rate)
- **Stability**: Handles edge cases including:
  - Variable-length sequences
  - Small context lengths (< 256 tokens)
  - Various head size and group size combinations
  - NaN prevention through proper masking in reduction operations

**Sample Test Output Structure**:
- Numerical comparison with reference implementations (PyTorch-based)
- Detailed error analysis for failing cases
- Performance metrics: execution time (μs/iter) and bandwidth (TB/s)
- Hash-based verification for regression detection

**Performance Comparison (Gluon vs ASM)**:
```
    context_length  batch_size num_heads  head_size  block_size           compute_type  query_length quant_mode  context_partition_size  trans_v  kv_varlen  use_aot_impl  quant_q  quant_kv  us_gluon  err_gluon  gluon_bandwith(TB/s) perf_gluon_vs_asm  us_asm  asm_bandwith(TB/s)
0             4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False    False     False      16.9          0                   0.5               NaN     NaN                 NaN
1             4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False    False     False      31.8          0                   0.2               NaN     NaN                 NaN
2             4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False    False     False     124.3          0                   2.0               NaN     NaN                 NaN
3             4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False    False     False     279.6          0                   0.9               NaN     NaN                 NaN
4             4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False    False     False      19.0          0                   0.4               NaN     NaN                 NaN
5             4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False    False     False      33.8          0                   0.2               NaN     NaN                 NaN
6             4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False    False     False     154.2          0                   1.6               NaN     NaN                 NaN
7             4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False    False     False     528.3          0                   0.5               NaN     NaN                 NaN
8             4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False    False     False      16.7          0                   0.5               NaN     NaN                 NaN
9             4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False    False     False      31.3          0                   0.2               NaN     NaN                 NaN
10            4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False    False     False      94.9          0                   2.6               NaN     NaN                 NaN
11            4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False    False     False     210.8          0                   1.2               NaN     NaN                 NaN
12            4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False    False     False      18.4          0                   0.4               NaN     NaN                 NaN
13            4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False    False     False      33.2          0                   0.2               NaN     NaN                 NaN
14            4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False    False     False     116.6          0                   2.1               NaN     NaN                 NaN
15            4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False    False     False     407.3          0                   0.6               NaN     NaN                 NaN
16            4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False     True      True      14.8          0                   0.3              223%    33.2                 0.1
17            4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False     True      True      33.0          0                   0.1              350%   115.3                 0.0
18            4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False     True      True      84.8          0                   1.4               73%    61.8                 2.0
19            4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False     True      True     254.9          0                   0.5               91%   233.0                 0.5
20            4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False     True      True      15.7          0                   0.2               NaN     NaN                 NaN
21            4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False     True      True      30.3          0                   0.1               NaN     NaN                 NaN
22            4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True      False         False     True      True      87.3          0                   1.4               NaN     NaN                 NaN
23            4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True      False         False     True      True     275.0          0                   0.5               NaN     NaN                 NaN
24            4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False     True      True      14.6          0                   0.3              203%    29.7                 0.1
25            4096           4   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False     True      True      31.7          0                   0.1              304%    96.6                 0.0
26            4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False     True      True      63.9          0                   1.9               87%    55.7                 2.2
27            4096         128   (16, 1)        128          16  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False     True      True     194.8          0                   0.6              103%   200.3                 0.6
28            4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False     True      True      15.6          0                   0.2               NaN     NaN                 NaN
29            4096           4   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False     True      True      29.9          0                   0.1               NaN     NaN                 NaN
30            4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             1  per_token                     256     True       True         False     True      True      66.3          0                   1.8               NaN     NaN                 NaN
31            4096         128   (16, 1)        128        1024  torch.float8_e4m3fnuz             4  per_token                     256     True       True         False     True      True     212.2          0                   0.6               NaN     NaN                 NaN
32            4096           4   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True      False         False    False     False      15.2          0                   0.5               NaN     NaN                 NaN
33            4096           4   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True      False         False    False     False      33.3          0                   0.2               NaN     NaN                 NaN
34            4096         128   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True      False         False    False     False     105.2          0                   2.3               NaN     NaN                 NaN
35            4096         128   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True      False         False    False     False     290.3          0                   0.9               NaN     NaN                 NaN
36            4096           4   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True      False         False    False     False      18.0          0                   0.4               NaN     NaN                 NaN
37            4096           4   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True      False         False    False     False      34.7          0                   0.2               NaN     NaN                 NaN
38            4096         128   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True      False         False    False     False     132.4          0                   1.9               NaN     NaN                 NaN
39            4096         128   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True      False         False    False     False     371.7          0                   0.7               NaN     NaN                 NaN
40            4096           4   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True       True         False    False     False      15.7          0                   0.5               NaN     NaN                 NaN
41            4096           4   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True       True         False    False     False      33.4          0                   0.2               NaN     NaN                 NaN
42            4096         128   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True       True         False    False     False      81.1          0                   3.0               NaN     NaN                 NaN
43            4096         128   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True       True         False    False     False     217.2          0                   1.1               NaN     NaN                 NaN
44            4096           4   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True       True         False    False     False      17.7          0                   0.4               NaN     NaN                 NaN
45            4096           4   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True       True         False    False     False      34.4          0                   0.2               NaN     NaN                 NaN
46            4096         128   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True       True         False    False     False      99.1          0                   2.5               NaN     NaN                 NaN
47            4096         128   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True       True         False    False     False     281.9          0                   0.9               NaN     NaN                 NaN
48            4096           4   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True      False         False     True      True      16.5          0                   0.2              201%    33.2                 0.1
49            4096           4   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True      False         False     True      True      40.9          0                   0.1              282%   115.4                 0.0
50            4096         128   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True      False         False     True      True     113.3          0                   1.1               55%    61.9                 2.0
51            4096         128   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True      False         False     True      True     374.1          0                   0.3               62%   233.0                 0.5
52            4096           4   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True      False         False     True      True      17.7          0                   0.2               NaN     NaN                 NaN
53            4096           4   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True      False         False     True      True      34.3          0                   0.1               NaN     NaN                 NaN
54            4096         128   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True      False         False     True      True     119.5          0                   1.0               NaN     NaN                 NaN
55            4096         128   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True      False         False     True      True     371.0          0                   0.3               NaN     NaN                 NaN
56            4096           4   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True       True         False     True      True      16.2          0                   0.2              183%    29.6                 0.1
57            4096           4   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True       True         False     True      True      39.2          0                   0.1              246%    96.6                 0.0
58            4096         128   (16, 1)        128          16         torch.bfloat16             1  per_token                     256     True       True         False     True      True      88.4          0                   1.4               63%    55.7                 2.2
59            4096         128   (16, 1)        128          16         torch.bfloat16             4  per_token                     256     True       True         False     True      True     278.0          0                   0.4               72%   200.1                 0.6
60            4096           4   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True       True         False     True      True      17.6          0                   0.2               NaN     NaN                 NaN
61            4096           4   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True       True         False     True      True      33.8          0                   0.1               NaN     NaN                 NaN
62            4096         128   (16, 1)        128        1024         torch.bfloat16             1  per_token                     256     True       True         False     True      True      91.8          0                   1.3               NaN     NaN                 NaN
63            4096         128   (16, 1)        128        1024         torch.bfloat16             4  per_token                     256     True       True         False     True      True     280.7          0                   0.4               NaN     NaN                 NaN
```

### Submission Checklist

- [x] Core kernel implementation with Gluon API
- [x] Support for FP8/FP16/BF16 data types
- [x] Both per-token and per-tensor quantization modes
- [x] Paged KV cache with multiple block size options
- [x] Reduction kernels for multi-partition contexts
- [x] AOT compilation infrastructure and tooling
- [x] Comprehensive test suite with reference validation
- [x] Documentation via extensive inline comments and docstrings
- [x] C++ wrapper templates for deployment integration
- [x] Performance benchmarking utilities
- [x] Fallback support for environments without Gluon (graceful degradation)

### CODE REVIEW

File: csrc/cpp_itfs/pa_gluon_aot/pa_decode_reduce_kernel.cpp.jinja [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#include "{{triton_header}}"
+
+
+extern "C" {
+void {{func_name}}(void* output_ptr,             // [num_seqs, num_kv_heads, q_group_size, head_size]
```
Comment (fsx950223) [inline csrc/cpp_itfs/pa_gluon_aot/pa_decode_reduce_kernel.cpp.jinja:5]: The file is used?

Comment (yanguahe) [reply csrc/cpp_itfs/pa_gluon_aot/pa_decode_reduce_kernel.cpp.jinja:5]: this is for test reduce kernel separately

File: csrc/cpp_itfs/gluon_aot_tools/extra/hip/compile.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
```
Comment (jwu10003) [inline csrc/cpp_itfs/gluon_aot_tools/extra/hip/compile.cpp]: 这个 compile 逻辑看看是不是和 triton.tools.compile.compile_kernel 重复了？
